### PR TITLE
dev/core#2056 Only retrieve pcp & soft_credit info when needed

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Currently we do 2 queries per contribution create to retrieve pcp & soft credit info. However, the
results of the queries are only used if pcp or soft credit keys exist in the params array

Before
----------------------------------------
The following queries run every contribution create but the results of the queries are only used on those contribution.create calls which pass pcp or soft_credit data

examples

|timestamp|query|seconds|rows found|columns requested|
|----------|-------|-------|-------|-------|
| 15/09/20 2:54 |  SELECT id FROM  civicrm_contribution_soft WHERE contribution_id = 49769031 AND pcp_id IS NOT NULL | 0.000624 | 0 | 0 |
| 15/09/20 2:54 |  SELECT id FROM  civicrm_contribution_soft WHERE contribution_id = 49769031 AND pcp_id IS NULL     | 0.000616 | 0 | 0 |

After
----------------------------------------
2 less queries per contribution create, unless needed.

Technical Details
----------------------------------------
Basically an extraction & wrapping in an if

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

File: CRM/Contribute/BAO/ContributionSoft.php [hunk 1]
```diff
@@ -53,39 +53,17 @@ public static function add(&$params) {
...
    * @param array $params
-   * @param object $contribution CRM_Contribute_DAO_Contribution
+   * @param CRM_Contribute_BAO_Contribution $contribution
    *
+   * @throws \CRM_Core_Exception
+   * @throws \CiviCRM_API3_Exception
    */
   public static function processSoftContribution($params, $contribution) {
-    //retrieve existing soft-credit and pcp id(s) if any against $contribution
-    $softIDs = self::getSoftCreditIds($contribution->id);
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/ContributionSoft.php:61]: @colemanw any chance you can look at this PR - it really just moves these 2 lines ^^ inside the if blocks below (so we don't do the query if we don't care about the answer)

**Comment by mepps** [comment]
I tested this on a local setup and saw no bugs.

**Comment by eileenmcnaughton** [comment]
@colemanw are you OK to merge this based on @mepps testing?

**Comment by seamuslee001** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
@seamuslee001 it worked this time!
